R/summary_funs.R

```diff
@@ -406,92 +417,114 @@ get_stat <- function(mu, lppd, y_wobs_test, stat, mu.bs = NULL, lppd.bs = NULL,
+        value <- srs_diffe$y_hat / n_full
+        # combine estimates of var(y_hat) and var(y)
+        value_se <- sqrt(srs_diffe$v_y_hat + srs_diffe$hat_v_y) / n_full
       } else {
-        value <- mean(wcv * crrct, na.rm = TRUE)
-        value.se <- weighted.sd(crrct, wcv, na.rm = TRUE) / sqrt(n_notna)
+        # full LOO estimator
+        value <- mean(wobs * correct) - mean(wobs * correct_baseline)
+        value_se <- .weighted_sd(correct - correct_baseline, wobs) / sqrt(n_full)
       }
     } else if (stat == "auc") {
-      if (!is.null(mu.bs)) {
-        auc.data <- cbind(y, mu, wcv)
-        auc.data.bs <- cbind(y, mu.bs, wcv)
-        value <- auc(auc.data) - auc(auc.data.bs)
-        idxs_cols <- seq_len(ncol(auc.data))
-        idxs_cols_bs <- setdiff(seq_len(ncol(auc.data) + ncol(auc.data.bs)),
+      if (n_loo < n_full) {
+        # subsampling LOO with AUC not implemented. Using fast LOO result.
+        mu <- summaries_fast$mu
+      }
+      if (!is.null(mu_baseline)) {
+        auc_data <- cbind(y, mu, wobs)
+        auc_data_baseline <- cbind(y, mu_baseline, wobs)
+        value <- .auc(auc_data) - .auc(auc_data_baseline)
+        idxs_cols <- seq_len(ncol(auc_data))
+        idxs_cols_bs <- setdiff(seq_len(ncol(auc_data) + ncol(auc_data_baseline)),
                                 idxs_cols)
         diffvalue.bootstrap <- bootstrap(
-          cbind(auc.data, auc.data.bs),
+          cbind(auc_data, auc_data_baseline),
           function(x) {
-            auc(x[, idxs_cols, drop = FALSE]) -
-              auc(x[, idxs_cols_bs, drop = FALSE])
+            .auc(x[, idxs_cols, drop = FALSE]) -
+              .auc(x[, idxs_cols_bs, drop = FALSE])
           },
           ...
         )
-        value.se <- sd(diffvalue.bootstrap, na.rm = TRUE)
+        value_se <- sd(diffvalue.bootstrap, na.rm = TRUE)
         lq_uq <- quantile(diffvalue.bootstrap,
                           probs = c(alpha_half, one_minus_alpha_half),
                           names = FALSE, na.rm = TRUE)
       } else {
-        auc.data <- cbind(y, mu, wcv)
-        value <- auc(auc.data)
-        value.bootstrap <- bootstrap(auc.data, auc, ...)
-        value.se <- sd(value.bootstrap, na.rm = TRUE)
+        auc_data <- cbind(y, mu, wobs)
+        value <- .auc(auc_data)
+        value.bootstrap <- bootstrap(auc_data, .auc, ...)
+        value_se <- sd(value.bootstrap, na.rm = TRUE)
         lq_uq <- quantile(value.bootstrap,
                           probs = c(alpha_half, one_minus_alpha_half),
                           names = FALSE, na.rm = TRUE)
       }
     }
   }
 
-  if (!stat %in% c("rmse", "auc")) {
-    lq <- qnorm(alpha_half, mean = value, sd = value.se)
-    uq <- qnorm(one_minus_alpha_half, mean = value, sd = value.se)
-  } else {
+  if (stat %in% c("mse","rmse") && is.null(mu_baseline)) {
+    # Compute mean and variance in log scale by matching the variance of a
+    # log-normal approximation
```

Comment:
Is it common to assume a log-normal distribution as the sampling distribution of the MSE and RMSE estimators? I haven't seen that yet (I think), but it might be perfectly fine (motivated by the central limit theorem, I guess).